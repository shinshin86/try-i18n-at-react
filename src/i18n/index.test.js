const en = require('./en.yml');
const ja = require('./ja.yml');
const ko = require('./ko.yml');

const getBlanks = obj => Object.keys(obj).filter(v => obj[v] === '');
const getAllKeys = obj => Object.keys(obj);

test('snapshot [en]', () => {
  expect(en).toMatchSnapshot();
});
test('snapshot [ja]', () => {
  expect(ja).toMatchSnapshot();
});
test('snapshot [ko]', () => {
  expect(ko).toMatchSnapshot();
});
test('check whitelist [en]', () => {
  const blanks = getBlanks(en);
  expect(blanks).toMatchSnapshot();
});
test('check whitelist [ja]', () => {
  const blanks = getBlanks(ja);
  expect(blanks).toMatchSnapshot();
});
test('check whitelist [ko]', () => {
  const blanks = getBlanks(ko);
  expect(blanks).toMatchSnapshot();
});

test('check key equals', () => {
  const langList = [en, ja, ko];
  const allLangKeyList = [];
  for (var lang of langList) {
    allLangKeyList.push(getAllKeys(lang));
    if (allLangKeyList.length > 1) {
      expect(allLangKeyList[allLangKeyList.length - 1]).toEqual(
        allLangKeyList[allLangKeyList.length - 2]
      );
    }
  }
});
