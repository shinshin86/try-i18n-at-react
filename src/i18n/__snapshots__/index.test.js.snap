// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check whitelist [en] 1`] = `Array []`;

exports[`check whitelist [ja] 1`] = `Array []`;

exports[`check whitelist [ko] 1`] = `Array []`;

exports[`snapshot [en] 1`] = `
Object {
  "ChangeLocale.Btn": "Switch languages",
  "ChangeLocale.Msg": "Switch language to Japanese. Is it OK?",
  "Top.Message": "Hello! { name }!",
  "Top.Title": "Sample Title",
}
`;

exports[`snapshot [ja] 1`] = `
Object {
  "ChangeLocale.Btn": "言語を切り替える",
  "ChangeLocale.Msg": "言語を韓国語に切り替えます。よろしいですか?",
  "Top.Message": "こんにちは! { name }!",
  "Top.Title": "サンプルタイトル",
}
`;

exports[`snapshot [ko] 1`] = `
Object {
  "ChangeLocale.Btn": "언어 전환",
  "ChangeLocale.Msg": "언어를 영어로 전환하십시오. 괜찮습니까?",
  "Top.Message": "여보세요! { name }!",
  "Top.Title": "샘플 제목",
}
`;
